### ISSUE TITLE
Cannot embed rich text directions into email template

### ISSUE THREAD

**Issue by ninetaLogik** [washingtontrails/vms#108]
This is related to story #31 in iteration 9. We were unable to embed the WorkParty.Trailhead. Directions in the Email Template. It is not possible to create a formula field to pull Directions onto Work Party because it is a rich text field. 
At this point we see 2 options: 
- Either put a link to the directions in the email template or
-  Create a rich text field on Work Party and sync it with directions on Trailhead via WF or trigger.

**Comment by mscholtz** [comment]
furthermore: we can't directly include the Directions in the template b/c it is two relationships away from the Registration.  Looks like we can only go one hop.  (or only traverse master-detail relationships)

we're also not sure if syncing w/ WF would work

another option would be to make directions plain text everywhere.

**Comment by WaBirder57** [comment]
We think it is pretty important for folks to have a “hard copy” of the directions so that they can access that info in a cached email on their phone when they are out of cell phone range.  We can’t count on them being able to access the info via a link when they really need it.

 

Charlie

**Comment by mscholtz** [comment]
Great, please mark this ticket as High Priority then, and we'll figure out
best approach soon.
But we didn't size this story assuming we were going to be tackling
something like this.

On Wed, May 20, 2015 at 4:49 PM, Charles Kahle <notifications@github.com>
wrote:

> We think it is pretty important for folks to have a “hard copy” of the
> directions so that they can access that info in a cached email on their
> phone when they are out of cell phone range. We can’t count on them being
> able to access the info via a link when they really need it.
>
>
>
> Charlie
>
>
>
> From: ninetaLogik [mailto:notifications@github.com]
> Sent: Wednesday, May 20, 2015 4:28 PM
> To: washingtontrails/vms
> Subject: [vms] Cannot embed rich text directions into email template (#108)
>
>
>
> This is related to story #31 <
> https://github.com/washingtontrails/vms/issues/31> in iteration 9. We
> were unable to embed the WorkParty.Trailhead. Directions in the Email
> Template. It is not possible to create a formula field to pull Directions
> onto Work Party because it is a rich text field.
> At this point we see 2 options:
>
> * Either put a link to the directions in the email template or
> * Create a rich text field on Work Party and sync it with directions on
> Trailhead via WF or trigger.
>
> —
> Reply to this email directly or view it on GitHub <
> https://github.com/washingtontrails/vms/issues/108> . <
> https://github.com/notifications/beacon/AH7VjpZumWlFcl1YuOh4oXnEFfoUyr-Fks5oLQ_zgaJpZM4Eik2u.gif>
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/washingtontrails/vms/issues/108#issuecomment-104073862>
> .
>



-- 

Matthew Scholtz
Percolator Consulting
(206) 289-0816 x3

http://percolatorconsulting.com/

**Comment by mscholtz** [comment]
@drewpiston could you actually look at this one first, before the other tickets?
I think first step is to talk through the options w/ Charlie and come up with a plan.  this is discussed in ticket comments.

**Comment by drewpiston** [comment]
Hi @WaBirder57 ,

I wanted to touch base with you this -- basically, because the Directions field on the Trailhead object isn't able to be pulled into the email template easily.

Our options are:

1. Maintain a separate Plain Text field set of directions.
2. Switch to only have Plain Text directions.
3. Write code to keep a copy of the Rich Text fields on the Work Party (or, possibly, on the WP Registration).  This would probably be fairly straightforward, in the 3-5 point range.
4. Switch to VF templates, which allow for more flexibility but are harder to maintain (requires comfort with html).

Sometimes VF Templates are great, but in this case I think it's overkill, so I'd recommend we go with #1 or #3 (unless #2 works well for you).

Happy to jump on the phone this afternoon if you'd like to discuss in detail.

 Drew

**Comment by drewpiston** [comment]
Talked with Charlie and we agreed on solution # 3 above.  We'll use triggers to push down the Directions rtf to either Work Party, WP Registration, or both so that it is available for email templates.

**Comment by mscholtz** [comment]
Sounds good.  Before writing the code let's create t the new rich text
field and test it on the template to make sure I it's gonna work there

**Comment by drewpiston** [comment]
This is ready for testing/reviewing!

**Comment by WaBirder57** [comment]
I tested and this is NOT working yet.  See ticket #174  with my testing procedure and results.

**Comment by drewpiston** [comment]
Left a comment on #174:
Hi @WaBirder57 ,

Sorry, I should have explained this more. I think that's actually a data issue.

Basically, what I've done for this ticket is create code to sync a new field on the Work Party (called Trailhead Directions) to the field on the Trailhead (called Directions). The code works fine for new records, but I neglected to do a data update for existing records. Since that Work Party was created before this code was put in place, the new Trailhead Directions field was blank.

I went ahead and make an update to that Work Party so that now the Trailhead Directions field is populated, and the email should work. I'll also update all the other records this morning.

Drew
